libshaderc_spvc/include/shaderc/spvc.h

```diff
@@ -0,0 +1,117 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SHADERC_SPVC_H_
+#define SHADERC_SPVC_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "libshaderc_util/common.h"
+#include "libshaderc_util/visibility.h"
+
+// An opaque handle to an object that manages all compiler state.
+typedef struct shaderc_spvc_compiler* shaderc_spvc_compiler_t;
+
+// Create a compiler.  A return of NULL indicates that there was an error.
+// Any function operating on a *_compiler_t must offer the basic
+// thread-safety guarantee.
+// [http://herbsutter.com/2014/01/13/gotw-95-solution-thread-safety-and-synchronization/]
+// That is: concurrent invocation of these functions on DIFFERENT objects needs
+// no synchronization; concurrent invocation of these functions on the SAME
+// object requires synchronization IF AND ONLY IF some of them take a non-const
+// argument.
+SHADERC_EXPORT shaderc_spvc_compiler_t shaderc_spvc_compiler_initialize(void);
+
+// Release resources.  After this the handle cannot be used.
+SHADERC_EXPORT void shaderc_spvc_compiler_release(shaderc_spvc_compiler_t);
+
+// An opaque handle to an object that manages options to a single compilation
+// result.
+typedef struct shaderc_spvc_compile_options* shaderc_spvc_compile_options_t;
+
+// Returns default compiler options.
+// A return of NULL indicates that there was an error initializing the options.
+// Any function operating on shaderc_spvc_compile_options_t must offer the
+// basic thread-safety guarantee.
+SHADERC_EXPORT shaderc_spvc_compile_options_t
+shaderc_spvc_compile_options_initialize(void);
+
+// Returns a copy of the given options.
+// If NULL is passed as the parameter the call is the same as
+// shaderc_spvc_compile_options_init.
+SHADERC_EXPORT shaderc_spvc_compile_options_t
+shaderc_spvc_compile_options_clone(
+    const shaderc_spvc_compile_options_t options);
+
+// Releases the compilation options. It is invalid to use the given
+// option object in any future calls. It is safe to pass
+// NULL to this function, and doing such will have no effect.
+SHADERC_EXPORT void shaderc_spvc_compile_options_release(
+    shaderc_spvc_compile_options_t options);
+
+// Sets the target shader environment, affecting which warnings or errors will
+// be issued.  The version will be for distinguishing between different versions
+// of the target environment.  The version value should be either 0 or
+// a value listed in shaderc_env_version.  The 0 value maps to Vulkan 1.0 if
+// |target| is Vulkan, and it maps to OpenGL 4.5 if |target| is OpenGL.
```

Comment:
Why do we want to have a value for "0"?

Reply:
see other comment about shaderc_spvc_compile_options_set_target_env()